### DISCUSSION TITLE
關於補數

### DISCUSSION THREAD

**Discussion by lx3211** [1101-GS4719-C/Python-Community-GS4719-C#454] [General]
老師這週上課的時候提到了補數的觀念
如果沒有修過相關課程可能會搞不清楚補數到底是用來做什麼的
**為什麼要多一個補數呢？**

電腦本身是由 0 跟 1 組合而成，沒有所謂「正負」的記號，**所有電腦的計算過程都是由「加法」組成**
而補數就是用來幫助電腦處理加減乘除用的
減法時以加補數方式進行，乘法則是連加、除法是連加補數

詳細內容網站有更多的介紹
裡面也有其他觀念可以參考
補數在第五項

[電腦工作原理](http://www.chwa.com.tw/TResource/HS/book1/ch2/ch2-3-1.htm)

**Comment by ThomasMeng0515** [comment]
減法在電腦運作上會以加上其二補數作運算，都是由加法所計算。
感謝分享~

**Comment by JunTingLin** [comment]
我也有寫一篇補數心得，有興趣大家可以參考💪
[💥 1補2補大亂鬥!補數整理與分享 #459](https://github.com/NCU-GS4719-Python/Python-Community/discussions/459)
